README.md

```diff
@@ -53,7 +53,47 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre. The
+following sections describe additional conventions followed in this project,
+extending the existing [PEP 8 Style Guide](https://peps.python.org/pep-0008/).
+
+#### The Complete Code Base:
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code conforms to [PEP 8](https://peps.python.org/pep-0008/).
+
+#### lib Tests (`tests/lib/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
+- define the expected behavior of the code for all use-cases of the different
+  consumers (i.e. CLI, demos).
+- all public functions and properties of all classes are tested with all
+  possible combinations of input:
```

Comment:
"all possible combinations" is a dangerous phrase. And impossible to meet in a literal sense.

Reply:
Yes, and yet I chose to write it like this anyways. :)

The truth is, currently we also do not have everywhere doc strings, type hinting, and all the other things of these categories.
I personally think it is more fitting to state the goals aimed at once clearly, instead of giving a too weak description.